.github/workflows/validate-migration.yaml

```diff
@@ -79,28 +69,58 @@ jobs:
+            CMAKE_CXX_FLAGS="-stdlib=libc++"
           ;;
               
           addressAndUndefinedSanitizer)
-            OPENSSL_FLAGS="-fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -nostdinc++ -nostdlib++ -stdlib=libc++ -fPIE"
-            CMAKE_CXX_FLAGS="-fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all"
+            BUILD_LIBCXX="true"
+            OPENSSL_FLAGS="-fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -nostdinc++ -nostdlib++ -stdlib=libc++ -fPIE -DPEDANTIC"
+            CMAKE_CXX_FLAGS="-fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -nostdinc++ -nostdlib++ -stdlib=libc++ -fPIE"
             CMAKE_EXE_LINKER_FLAGS="-lc++ -lc++abi -lunwind -pie"
             LLVM_USE_SANITIZER="Address;Undefined"
+          
+            # Use Clang's Sanitizer implementation
+            CC="sccache clang-16"
+            CXX="sccache clang++-16"
+            CMAKE_CXX_COMPILER="clang++-16"
+            CMAKE_C_COMPILER="clang-16"
           ;;
         
           threadSanitizer)
-            OPENSSL_FLAGS="-fsanitize=thread -fno-sanitize-recover=all -nostdinc++ -nostdlib++ -stdlib=libc++"
-            CMAKE_CXX_FLAGS="-fsanitize=thread -fno-sanitize-recover=all"
+            BUILD_LIBCXX="true"
+            OPENSSL_FLAGS="-fsanitize=thread -fno-sanitize-recover=all -nostdinc++ -nostdlib++ -fPIE -stdlib=libc++"
+            CMAKE_CXX_FLAGS="-fsanitize=thread -fno-sanitize-recover=all -nostdinc++ -nostdlib++ -fPIE -stdlib=libc++"
             CMAKE_EXE_LINKER_FLAGS="-lc++ -lc++abi -lunwind -pie"
             LLVM_USE_SANITIZER="Thread"
+          
+            # Use Clang's Sanitizer implementation
+            CC="sccache clang-16"
+            CXX="sccache clang++-16"
+            CMAKE_CXX_COMPILER="clang++-16"
+            CMAKE_C_COMPILER="clang-16"
           ;;
         
           valgrindMemcheck)
+            OPENSSL_FLAGS="-DPURIFY" # See https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/rand/md_rand.c#L490-L500
             VALGRIND_FLAGS="--tool=memcheck --leak-check=full --show-leak-kinds=all --track-origins=yes"
+          
+            # Use GCC instead of Clang for Valgrind since Ubuntu ships with suppression files for glibc
+            CC="sccache gcc"
+            CXX="sccache g++"
+            CMAKE_CXX_COMPILER="g++"
+            CMAKE_C_COMPILER="gcc"
           ;;
         
           valgrindHelgrind)
+            OPENSSL_FLAGS="-DPURIFY" # See https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/rand/md_rand.c#L490-L500
             VALGRIND_FLAGS="--tool=helgrind --track-lockorders=yes --history-level=full" 
+          
+            # Use GCC instead of Clang for Valgrind since Ubuntu ships with suppression files for glibc
+
+            CC="sccache gcc"
+            CXX="sccache g++"
+            CMAKE_CXX_COMPILER="g++"
+            CMAKE_C_COMPILER="gcc"
```

Comment:
Instead, of scattering the exact gcc/g++ setting in multiple places, I suggest setting a `P4FUSION_USE_GCC=1` here. Then you can do an `if` condition on that near the `echo` lines and set `CC`, `CXX`, `CMAKE_CXX_COMPILER` and `CMAKE_C_COMPILER` accordingly.